Source/Core/Core/AchievementManager.cpp

```diff
@@ -830,100 +303,110 @@ const AchievementManager::BadgeStatus& AchievementManager::GetGameBadge() const
   return m_game_badge;
 }
 
-const AchievementManager::UnlockStatus&
-AchievementManager::GetUnlockStatus(AchievementId achievement_id) const
+const AchievementManager::BadgeStatus& AchievementManager::GetAchievementBadge(AchievementId id,
+                                                                               bool locked) const
 {
-  return m_unlock_map.at(achievement_id);
+  if (locked)
+    return m_locked_badges.at(id);
```

Comment:
Worth mentioning `at` throws if the id isn't available.  Maybe consider `[]` which would create the element instead?